File: src/plot/run_plots.jl
using Pkg

cd(@__DIR__)
# TODO create julia project in tps_eval
Pkg.activate("../../../../terpene_generation/src") # TODO delete
#Pkg.instantiate()

# --- Script ---------------------------------------------------------
include("plot_comparison.jl")

num_args = length(ARGS)

if num_args == 4 || num_args == 5
    fasta_paths = split(ARGS[1], ",")
    data_names = split(ARGS[2], ",")
    data_colors = Symbol.(split(ARGS[3], ","))
    targets = split(ARGS[4], ",")
    save_dir = num_args == 5 ? ARGS[5] : nothing
else
    error("Invalid number of arguments. Expected 4 or 5, got $num_args.")
end

fasta_paths = Vector{String}(fasta_paths)
data_names = Vector{String}(data_names)
data_colors = Vector{Symbol}(data_colors)
targets = Vector{String}(targets)

plot_comparison(
    fasta_paths,
    data_names,
    data_colors,
    targets;
    save_dir=save_dir
)

File: src/plot/run_plots.py
from __future__ import annotations

import sys
from pathlib import Path

CURRENT_DIR = Path(__file__).resolve().parent
SRC_DIR = CURRENT_DIR.parent
if str(SRC_DIR) not in sys.path:
    sys.path.insert(0, str(SRC_DIR))

from plot.plot_comparison import plot_comparison  # noqa: E402


def main(argv: list[str]) -> None:
    num_args = len(argv)
    if num_args < 3:
        raise SystemExit(
            "Invalid number of arguments. Expected at least 3: "
            "fasta_paths, data_names, data_colors."
        )

    fasta_paths = argv[0].split(",")
    data_names = argv[1].split(",")
    data_colors = argv[2].split(",")

    kwargs: dict = {}
    i = 3
    while i < num_args:
        flag = argv[i]
        if flag == "--targets":
            if i + 1 >= num_args:
                raise SystemExit("Missing value for --targets.")
            kwargs["targets"] = argv[i + 1].split(",")
            i += 2
        elif flag == "--save_dir":
            if i + 1 >= num_args:
                raise SystemExit("Missing value for --save_dir.")
            kwargs["save_dir"] = argv[i + 1]
            i += 2
        else:
            raise SystemExit(f"Unknown argument: {flag}")

    plot_comparison(fasta_paths, data_names, data_colors, **kwargs)


if __name__ == "__main__":
    main(sys.argv[1:])
